tests/integration/targets/postgresql_privs/tasks/postgresql_privs_general.yml

```diff
@@ -683,6 +683,64 @@
     - result.rowcount == 0
   when: postgres_version_resp.stdout is version('10', '>=')
 
+# Test
+- name: Grant execute with grant option on pg_create_restore_point function
+  postgresql_privs:
+    privs: EXECUTE
+    type: function
+    schema: pg_catalog
+    obj: pg_create_restore_point(text)
+    db: "{{ db_name }}"
+    roles: "{{ db_user2 }}"
+    grant_option: yes
+    state: present
+  register: result
+
+# Checks
+- assert:
+    that: result is changed
+
+- name: Check that user has GRANT privilege on the function
+  postgresql_query:
+    query: SELECT proacl FROM pg_proc WHERE proname='pg_create_restore_point'
+    db: "{{ db_name }}"
+    login_user: "{{ db_user2 }}"
+    login_password: password
+  become: yes
+  become_user: "{{ pg_user }}"
+  register: result
+
+- assert:
+    that: "'{{ db_user2 }}=X*/postgres' in result.query_result[0].proacl"
+
+# Test
+- name: Grant execute without specifying grant_option to check idempotence
+  postgresql_privs:
+    privs: EXECUTE
+    type: function
+    schema: pg_catalog
+    obj: pg_create_restore_point(text)
+    db: "{{ db_name }}"
+    roles: "{{ db_user2 }}"
+    state: present
+  register: result
+
+# Checks
+- assert:
+    that: result is not changed
+
+- name: Check that user has GRANT privilege on the function
```

Comment:
I guess I knew, why the tests are failing. Pls remove the GRANT privilege after your tests/checks from this user. Because in the cleanup in L1147 the user cannot be removed.

Reply:
Only GRANT or both, GRANT and EXECUTE?